assignment3.txt

```diff
@@ -0,0 +1,7 @@
+1. The code is not valid as the syntax for the for loop is incorrect. First, i is never declared before it is used and is not given a starting value. Inside the loop, there is also no integer in the printf statement. 
+
+2. The variable i should not be reused within the loop because the computer will not be able to differentiate between the i being used to increment in the loop and the i in loop.
+
+3. i. O(2n) -- worst case scenario it would be O(n) for each loop --> O(2n)
+   ii. O(n^2)
```

Comment:
The answer is actually O(n^4) - again, for loops that run from 0 to n have the time complexity O(n). So the top loop is O(n), the middle Loop is O(n^2) iterations because it iterates from 1 to i^2. The if statement is O(1 / n) times because of the modulo function. Lastly, the inner loop  is O(n^2) iterations because it iterates from 0 to j and j = i^2 = n^2. 

Therefore, the sum = -(1/12)*n + (3/8)*n^2 - (5/12)*n^3 + (1/8)*n^4  = O(n^4)